pkg/cloud/aliyunprovider.go

```diff
@@ -842,3 +919,52 @@ func generateSlimK8sNodeFromV1Node(node *v1.Node) *SlimK8sNode {
 
 	return NewSlimK8sNode(instanceType, regionID, priceUnit, memorySizeInKiB, osType, providerID, instanceFamily, IsIoOptimized)
 }
+
+// generateSlimK8sDiskFromV1PV function generates SlimK8sDisk from v1.PersistentVolume and DescribeDisk API(If required) of alibaba
+// to generate slim disk type that can be used to fetch pricing information.
+func generateSlimK8sDiskFromV1PV(pv *v1.PersistentVolume, regionID string) *SlimK8sDisk {
+	//Regular expression to get the GiB storage size for the API call
+	sizeRegEx := regexp.MustCompile("(.*?)Gi")
+
+	// All PVs are data disks while local disk are categorized as system disk
+	diskType := ALIBABA_DATA_DISK_CATEGORY
+
+	//TO-DO: Disk supports month and hour prices , defaulting to hour
+	priceUnit := ALIBABA_HOUR_PRICE_UNIT
+
+	sizeQuantity := fmt.Sprintf("%s", pv.Spec.Capacity.Storage())
+
+	res := sizeRegEx.FindAllStringSubmatch(sizeQuantity, 1)
+
+	// This is the default value used for the DescribePrice DataDisk size, if any error occured defaulting it to 2000GiB's price
+	sizeInGiB := "2000"
+	if len(res) != 0 {
+		sizeInGiB = res[0][1]
+	}
```

Comment:
This could panic, I think: https://go.dev/play/p/lHfBVsr2nSM

Reply:
Very unlikely scenario to occur since Ali cloud clearly states on UI atleast to create 20 Gi PVs and doesnt allow, but I went ahead and incorporated the ask and unit test cases around it!
![Screen Shot 2022-11-23 at 3 14 11 PM](https://user-images.githubusercontent.com/11470561/203661935-fa0a4f38-df6a-49ab-9368-a2d50faf7d00.png)